sotodlib/preprocess/preprocess_util.py

```diff
@@ -406,64 +599,123 @@ def preproc_or_load_group(obs_id, configs, dets, logger=None,
+                        proc_aman.save(outputs_proc['temp_file'], outputs_proc['db_data']['dataset'], overwrite)
+                    else:
+                        logger.info(f"{outputs_proc['temp_file']}:{outputs_proc['db_data']['dataset']} already exists.")
+
+                except Exception as e:
+                    error = f'Failed to load: {obs_id} {dets}'
+                    errmsg = f'{type(e)}: {e}'
+                    tb = ''.join(traceback.format_tb(e.__traceback__))
+                    logger.info(f"{error}\n{errmsg}\n{tb}")
+                    return error, [errmsg, tb], None
+                if success != 'end':
+                    # If a single group fails we don't log anywhere just mis an entry in the db.
+                    return success, [obs_id, dets], [obs_id, dets], None
+
+                return success, [obs_id, dets], outputs_proc, aman
     else:
+        # pipeline for init config
         logger.info(f"Generating new preproc db entry for {obs_id} {dets}")
-        pipe = Pipeline(configs["process_pipe"], plot_dir=configs["plot_dir"], logger=logger)
+        pipe_init = Pipeline(configs_init["process_pipe"], plot_dir=configs_init["plot_dir"], logger=logger)
         try:
-            aman = context.get_obs(obs_id, dets=dets)
-            tags = np.array(context.obsdb.get(aman.obs_info.obs_id, tags=True)['tags'])
+            aman = context_init.get_obs(obs_id, dets=dets)
+            tags = np.array(context_init.obsdb.get(aman.obs_info.obs_id, tags=True)['tags'])
             aman.wrap('tags', tags)
-            proc_aman, success = pipe.run(aman)
+            proc_aman, success = pipe_init.run(aman)
             aman.wrap('preprocess', proc_aman)
         except Exception as e:
             error = f'Failed to load: {obs_id} {dets}'
             errmsg = f'{type(e)}: {e}'
             tb = ''.join(traceback.format_tb(e.__traceback__))
             logger.info(f"{error}\n{errmsg}\n{tb}")
-            return error, [errmsg, tb], None
+            return error, [errmsg, tb], [errmsg, tb], None
         if success != 'end':
             # If a single group fails we don't log anywhere just mis an entry in the db.
-            return success, [obs_id, dets], None
-        newpath = f'temp/{obs_id}'
-        for cg in cur_groups[0]:
-            newpath += f'_{cg}'
-        temp_config = swap_archive(configs, newpath+'.h5')
-        policy = ArchivePolicy.from_params(temp_config['archive']['policy'])
-        dest_file, dest_dataset = policy.get_dest(obs_id)
-        for gb, g in zip(group_by, cur_groups[0]):
-            if gb == 'detset':
-                dest_dataset += "_" + g
+            return success, [obs_id, dets], [obs_id, dets], None
+
+        outputs_init = save_group(obs_id, configs_init, dets, context_init, subdir='temp')
+        if overwrite or not os.path.exists(outputs_init['temp_file']):
```

Comment:
This check should probably happen before getting to this step as referenced above in L642